stomp_moveit/src/stomp_smoothing_adapter.cpp

```diff
@@ -0,0 +1,139 @@
+
+// ROS
+#include <ros/ros.h>
+#include <class_loader/class_loader.hpp>
+
+// MoveIt
+#include <moveit/planning_request_adapter/planning_request_adapter.h>
+#include <moveit/planning_interface/planning_interface.h>
+
+// STOMP
+#include <stomp_moveit/stomp_planner.h>
+
+namespace stomp_moveit
+{
+class StompSmoothingAdapter : public planning_request_adapter::PlanningRequestAdapter
+{
+public:
+  StompSmoothingAdapter() : planning_request_adapter::PlanningRequestAdapter()
+  {
+  }
+
+  virtual void initialize(const ros::NodeHandle& node_handle) override
+  {
+    ros::NodeHandle nh(node_handle);
+    if (!StompPlanner::getConfigData(nh, group_config_))
+      ROS_ERROR("Unable to find valid group config for StompSmoothingAdapter");
+  }
+
+  virtual std::string getDescription() const override
+  {
+    return "Stomp Smoothing Adapter";
+  }
+
+  bool adaptAndPlan(const PlannerFn& planner, const planning_scene::PlanningSceneConstPtr& ps,
+                    const planning_interface::MotionPlanRequest& req, planning_interface::MotionPlanResponse& res,
+                    std::vector<std::size_t>& added_path_index) const override
+  {
+    // Following call to planner() calls the OMPL planner and stores the trajectory inside the MotionPlanResponse res
+    // variable which is then used by STOMP for optimization of the computed trajectory
+    if (!planner(ps, req, res))
+      return false;
+
+    // STOMP reads the seed trajectory from trajectory constraints so we need to convert the waypoints first
+    const size_t seed_waypoint_count = res.trajectory_->getWayPointCount();
+    const std::vector<std::string> joint_names =
+      res.trajectory_->getFirstWayPoint().getJointModelGroup(req.group_name)->getActiveJointModelNames();
+    const size_t joint_count = joint_names.size();
+    planning_interface::MotionPlanRequest seed_req = req;
+    seed_req.trajectory_constraints.constraints.clear();
+    seed_req.trajectory_constraints.constraints.resize(seed_waypoint_count);
+    for (size_t i = 0; i < seed_waypoint_count; ++i)
+    {
+      seed_req.trajectory_constraints.constraints[i].joint_constraints.resize(joint_count);
+      for (size_t j = 0; j < joint_count; ++j)
+      {
+        seed_req.trajectory_constraints.constraints[i].joint_constraints[j].joint_name = joint_names[j];
+        seed_req.trajectory_constraints.constraints[i].joint_constraints[j].position =
+          res.trajectory_->getWayPoint(i).getVariablePosition(joint_names[j]);
+      }
+    }
+
+    // Get group config
+    const auto& group_config_it = group_config_.find(req.group_name);
+    if (group_config_it == group_config_.end())
+    {
+      ROS_ERROR_STREAM("STOMP is not configured for planning group " << req.group_name);
+      res.error_code_.val = moveit_msgs::MoveItErrorCodes::FAILURE;
+      return false;
+    }
+
+    // Initialize STOMP Planner
+    stomp_moveit::StompPlanner stompPlanner(req.group_name, group_config_it->second, ps->getRobotModel());
```

Comment:
variable name should be [under_scored](https://google.github.io/styleguide/cppguide.html#Variable_Names)